assembly/src/assembler/linker/mod.rs

```diff
@@ -419,6 +419,26 @@ impl Linker {
         Ok(module_indices)
     }
 
+    /// Links `kernel` using the current state of the linker.
+    ///
+    /// Returns the module index of the kernel module, which is expected to provide the public
+    /// interface of the final assembled kernel.
+    ///
+    /// This differs from `link` in that we allow all AST modules in the module graph access to
+    /// kernel features, e.g. `caller`, as if they are defined by the kernel module itself.
+    pub fn link_kernel(&mut self, kernel: Box<Module>) -> Result<Vec<ModuleIndex>, LinkerError> {
```

Comment:
nit: it feels a bit weird that we return `Vec<ModuleIndex>` from here rather than just `ModuleIndex` (especially since in the comments we say that we return a single module index). Maybe a slightly cleaner way would be to return a single module index and convert it into a vector at the call site?

Reply:
Do we anticipate a future where we can provide multiple modules that export syscalls, or only ever one? If the former, we should keep this as-is, otherwise we can change it as suggested.

Reply:
I guess it could be convenient to have multiple top-level modules for code organization purposes (our current top level module for transaction kernel is over 1K lines). So, let's keep as is.